pal/inc/CCSpinLock.hpp

```diff
@@ -0,0 +1,34 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#ifndef CC_PAL_INC_CCSPINLOCK_H
+#define CC_PAL_INC_CCSPINLOCK_H
+
+class CCSpinLock
+{
+    unsigned int  enterCount;
+    bool          shouldTrackThreadId;
+    char          lock;
+    size_t        threadId;
```

Comment:
no particular reason. They both do the same on xplat yet agree DWORD should be the type there.